memo/memo_workflow.go

```diff
@@ -0,0 +1,144 @@
+package memo
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"go.temporal.io/api/common/v1"
+	workflowpb "go.temporal.io/api/workflow/v1"
+	"go.temporal.io/sdk/activity"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/converter"
+	"go.temporal.io/sdk/log"
+	"go.temporal.io/sdk/workflow"
+)
+
+/*
+ * This sample shows how to use memo. (Note this feature only work with Elasticsearch)
+ */
+
+// ClientContextKey is the key for lookup
+type ClientContextKey struct{}
+
+const (
+	// namespace used for this sample. "default" namespace always exists on the server.
+	namespace = "default"
+)
+
+var (
+	// ErrClientNotFound when client is not found on context.
+	ErrClientNotFound = errors.New("failed to retrieve client from context")
+	// ClientCtxKey for retrieving client from context.
+	ClientCtxKey ClientContextKey
+)
+
+// MemoWorkflow workflow definition
+func MemoWorkflow(ctx workflow.Context) error {
+	logger := workflow.GetLogger(ctx)
+	logger.Info("Memo workflow started")
+
+	// Get memo that were provided when workflow was started.
+	info := workflow.GetInfo(ctx)
+	val := info.Memo.Fields["description"]
+	var currentDescription string
+	err := converter.GetDefaultDataConverter().FromPayload(val, &currentDescription)
+	if err != nil {
+		logger.Error("Get memo failed.", "Error", err)
+		return err
+	}
+	logger.Info("Current memo value.", "description", currentDescription)
+
+	// Update memo.
+	memo := map[string]interface{}{
+		"description": "Test upsert memo workflow",
+	}
+	// This won't persist memo on server because commands are not sent to server,
+	// but local cache will be updated.
+	err = workflow.UpsertMemo(ctx, memo)
+	if err != nil {
+		return err
+	}
+
+	// Print current memo with modifications above.
+	info = workflow.GetInfo(ctx)
+	err = printMemo(info.Memo, logger)
+	if err != nil {
+		return err
+	}
+
+	// Now send commands to the server and let Elasticsearch update the index.
```

Comment:
Unrelated to Elasticsearch. Also, I think it may be confusing to users that they have to sleep to have something sent to the server. I think that information can be left off (clearly this is a copy/paste from the old `searchattributes` example, but it may not be the best).

Also, we encourage people to ignore errors.

Reply:
Comments are fixed.